src/components/application_manager/include/application_manager/message_helper.h

```diff
@@ -259,6 +259,23 @@ class MessageHelper {
   static smart_objects::SmartObjectList CreateAddCommandRequestToHMI(
       ApplicationConstSharedPtr app, ApplicationManager& app_mngr);
 
+  static smart_objects::SmartObjectSPtr CreateMessageForHMI(
+      hmi_apis::messageType::eType message_type, const uint32_t correlation_id);
+
+  /**
+   * @brief Creates Message for sending to HMI for unsubscribe from specified
+   * VehicleData. Creates and returns message with vehicle data for unsubscribe.
+   * @param vehicle_data - collection with vehicle data types to unsubscribe.
+   * @param app - application which should be unsubscribed from specified
+   * vehicle data.
+   * @return - request filled with vehicle data params to unsubscribe
+   * ,application id and function id.
+   */
+  static smart_objects::SmartObjectSPtr
+  CreateUnsubscribeVehicleDataMessageForHMI(
+      const VehicleInfoSubscriptions& vehicle_data,
+      const application_manager::ApplicationSharedPtr& app);
```

Comment:
@IGapchuk any reasons to pass shared pointer by reference?

Reply:
@AKalinich-Luxoft Fix in 11b04298dca9cf017301e7cf54e3b206cd43d883.